artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -345,16 +357,19 @@ public void disconnect(boolean criticalError) {
    }
 
    @Override
-   public boolean checkDataReceived() {
-      boolean res = dataReceived;
-
-      dataReceived = false;
-
-      return res;
+   public void flush() {
+      checkInactivity();
    }
 
-   @Override
-   public void flush() {
+   private void checkInactivity() {
+      if (!this.useKeepAlive) {
```

Comment:
You still need to set dataReceived to false, otherwise the TTL from ConnectionFActory will kick in and this will be killed per inactivity.


TTL in should be done by the dataREcieved attribute from the super class.


Sending a TTL to the client should be a different thing. i think it's just a matter to send the TTL back to the client.



You have changed way more than needed I think.